src/mixins/hideLabelOnDrag.js

```diff
@@ -8,8 +8,9 @@ export default {
 
       this.shape.attr('label/display', 'none');
     },
-    showLabel() {
+    async showLabel() {
       this.shape.attr('label/display', 'initial');
+      await this.$nextTick();
```

Comment:
Going to add some context here as it wasn't clear right away why this fixes the problem:

When a boundary event is dragged off a task, it will snap back to its original task, or onto a new task.

When the mouse is released (`pointerup` event), it will show the label but also add an event listener right away, to hide the label again on position change. And we have a race condition!

The event listener is added before the boundary event snaps to that task, causing it to be triggered right away, hiding the label.

Reply:
Yup. Sometimes showLabel won, other times it was hideLabel. 👍